### ISSUE TITLE
test command in package.json fails

### ISSUE THREAD

**Issue by st-h** [ijlee2/ember-codemod-v1-to-v2#26]
After running this codemod the test-app `package.json` has the following script:

```
"test": "concurrently \"npm:lint\" \"npm:test:*\" --names \"lint,test:\"",
```

when I run `npm run test` in add-on root, I see the following error:

```
[test:ember-compatibility] Error!
[test:ember-compatibility] Error: Source and destination must not be the same.
[test:ember-compatibility] Error: Source and destination must not be the same.
[test:ember-compatibility]     at /Users/steve/code/ember/ember-container-query/node_modules/fs-extra/lib/util/stat.js:39:17
[test:ember-compatibility]     at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
```

I had a look at ember-container-query and noticed that the script is different:

```
"test": "ember test",
```

If I apply those changes to my add-on, `npm run test` works fine again, but it does not seem to execute the ember-try scenarios.

**How to reproduce**

If I modify `ember-container-query/test-app/package.json` to have the same content as my add-on after applying this codemod:

```
    "test": "ember test",
    "test:ember-compatibility": "../node_modules/.bin/ember try:all"
```

I am seeing the same error when running `npm run test` in the add-on root. 

I currently have no idea if this is an issue with the script this codemod generates or if this is an issue with embers tooling itself. Unfortunately searching for similar issues did not return any useful result so far.

**Comment by ijlee2** [comment]
@st-h Thanks for reporting the issue. I hadn't thought about the default setup that you (and likely others) have.

As you noticed, I had redefined the `test`, `test:ember`, and `test:ember-compatibility` scripts for `ember-container-query`. This is because I didn't agree with the scripts that `ember-cli` generates via blueprint.

As of `v4.12.1`, when we create an addon with `ember-cli`, we will get the following scripts: (source code [#1](https://github.com/ember-cli/ember-cli/blob/v4.12.1/blueprints/app/files/package.json#L25-L26), [#2](https://github.com/ember-cli/ember-cli/blob/v4.12.1/blueprints/addon/additional-package.json#L6))

```json5
/* package.json */
{
  "scripts": {
    "lint": "concurrently \"npm:lint:*(!fix)\" --names \"lint:\"",
    /* The lint:* scripts go here */
    "test": "concurrently \"npm:lint\" \"npm:test:*\" --names \"lint,test:\"",
    "test:ember": "ember test",
    "test:ember-compatibility": "ember try:each"
  }
}
```

The script above mean: By default, (1) the `test` script runs all `lint:*` checks and (2) `test:ember-compatibility` script all `ember-try` scenarios.

In my experience, it's better to separate linting and testing and to run only 1 `ember-try` scenario at a time. This helps us optimize continuous integration (CI), because we can more easily parallelize tasks to decrease runtime and cost.

That's why, the scripts in `ember-container-query` (in the `test-app`) look different. (source code [#3](https://github.com/ijlee2/ember-container-query/blob/4.0.0-alpha.6/test-app/package.json#L38-L39))

```json5
/* package.json */
{
  "scripts": {
    "lint": "concurrently \"npm:lint:*(!fix)\" --names \"lint:\"",
    /* The lint:* scripts go here */
    "test": "ember test",
    "test:ember-compatibility": "ember try:one"
  }
}
```

**Comment by ijlee2** [comment]
The `lint` and `test` scripts, which this codemod added to your root `package.json`, were derived from `addon-blueprint`. (source code [#4](https://github.com/embroider-build/addon-blueprint/blob/v1.5.0/src/root-package-json.js#L38-L121))

```json5
/* Workspace root's package.json (with yarn as an example) */
{
  "scripts": {
    "lint": "yarn workspaces run lint",
    "test": "yarn workspaces run test"
  }
}
```

These scripts mean: Run all `lint` and `test` scripts that you can find across all workspaces. Note, the root has no knowledge of what each workspace does for its own `lint` and `test` scripts. So it's possible that by running `lint` and `test` at the root level, we may run into issues in some workspace.

I'm guessing that [the comment in lines 114-119](https://github.com/embroider-build/addon-blueprint/blob/v1.5.0/src/root-package-json.js#L114-L119) had been added for a similar reason.

---

In short, I'll have to investigate how the `lint` and `test` scripts should look like for a v2 addon so that the default migration experience is less prone to errors. Thanks again for notifying me.

**Comment by st-h** [comment]
Thanks a lot for the explanation. I wish I had some time to get more involved with the embroider and v2 stuff. I was just updating an addon to v2 as I was trying to resolve the ember-modifier issue I am having with container query. I thought that would be the best way to address the auto-import dependency requirement, but turns out there are more gotchas than I expected.

**Comment by st-h** [comment]
Yes, I'll close this. Reasonable defaults should probably be best provided by the v2 spec / documentation and not by a codemod.
